internal/wordvault/server.go

```diff
@@ -777,6 +782,83 @@ func (s *Server) GetDailyLeaderboard(ctx context.Context, req *connect.Request[p
 	return connect.NewResponse(resp), nil
 }
 
+func (s *Server) GetFsrsParameters(ctx context.Context, req *connect.Request[pb.GetFsrsParametersRequest]) (
+	*connect.Response[pb.GetFsrsParametersResponse], error) {
+	user := auth.UserFromContext(ctx)
+	if user == nil {
+		return nil, unauthenticated("user not authenticated")
+	}
+
+	dbparams, err := s.fsrsParams(ctx, int64(user.DBID), nil)
+	if err != nil {
+		return nil, err
+	}
+
+	params := &pb.FsrsParameters{
+		Scheduler:        pb.FsrsScheduler_FSRS_SCHEDULER_LONG_TERM,
+		RequestRetention: dbparams.RequestRetention,
+	}
+
+	if dbparams.EnableShortTerm {
+		params.Scheduler = pb.FsrsScheduler_FSRS_SCHEDULER_SHORT_TERM
+	}
+
+	resp := &pb.GetFsrsParametersResponse{
+		Parameters: params,
+	}
+
+	return connect.NewResponse(resp), nil
+}
+
+func (s *Server) EditFsrsParameters(ctx context.Context, req *connect.Request[pb.EditFsrsParametersRequest]) (
+	*connect.Response[pb.EditFsrsParametersResponse], error) {
+	user := auth.UserFromContext(ctx)
+	if user == nil {
+		return nil, unauthenticated("user not authenticated")
+	}
+
+	if req.Msg.Parameters.RequestRetention < 0 || req.Msg.Parameters.RequestRetention > 1 {
+		return nil, invalidArgError("invalid retention value")
```

Comment:
@BenMusch One more change - sorry about the delay, btw! The FSRS paper/algorithms state that this works best with retention settings between 0.7 and 0.97. I would like to limit it to these values. We will probably get pathological behavior if it's allowed to be 0 or 1 - I can imagine divisions by zero and the like.

Reply:
See here https://github.com/open-spaced-repetition/fsrs4anki/blob/main/docs/tutorial.md (search for permissible range). The graph below is interesting and shows how the workload changes as the retention increases.

Reply:
done!